public/app/features/trails/otel/util.ts

```diff
@@ -191,8 +191,9 @@ export function limitOtelMatchTerms(
         jobsRegex += `${jobsList[i]}`;
         instancesRegex += `${instancesList[i]}`;
       } else {
-        jobsRegex += `|${jobsList[i]}`;
-        instancesRegex += `|${instancesList[i]}`;
+        // check to make sure we aren't duplicating job or instance
+        jobsRegex += !jobsRegex.includes(jobsList[i]) ? `|${jobsList[i]}` : '';
+        instancesRegex += !instancesRegex.includes(instancesList[i]) ? `|${instancesList[i]}` : '';
```

Comment:
```suggestion
        jobsRegex += jobsRegex.includes(jobsList[i]) ? '' : `|${jobsList[i]}`;
        instancesRegex += instancesRegex.includes(instancesList[i]) ? '' : `|${instancesList[i]}`;
```

Reply:
I think this way is easier to read. Feel free to ignore it if you disagree.

Reply:
Starting to read any conditionals with `!` is generally confusing. I changed it :)